.grit/patterns/hathora_to_new_ts_sdk.md

```diff
@@ -0,0 +1,226 @@
+# Upgrade Hathora to Dedicated TS SDK
+
+```
+engine marzano(0.1)
+language js
+
+pattern replace_resource_import($resource) {
+  contains import_statement(import = import_clause(name=named_imports($imports)), source= `"@hathora/hathora-cloud-sdk"`) where {
+    if ($imports <: contains $import where $import <: $resource) {
+      $sdk = `sdk`,
+      // TODO what's the best way to ask them to provide their SDK?
+      // emit a file with a TODO? what if they use relative imports?
+      // how do we know what import path to use? add that to the TODO?
+      $sdk <: ensure_import_from(source=`"./sdk"`),
+    } else {
+      true
+    }
+  }
+}
+
+// TODO finish these various resource-specific checks.
+// if it turns out they're very regular, you could just regex-and-recase them.
+function resource_type_import_source($resource) {
+  if($resource <: `RoomV1Api`) {
+    $result = `"@hathora/cloud-sdk-typescript/dist/sdk/roomv1"`
+  } else if ($resource <: `AuthV1Api`) {
+    $result = `"@hathora/cloud-sdk-typescript/dist/sdk/authv1"`
+  } else if ($resource <: `LobbyV1Api`) {
+    $result = `"@hathora/cloud-sdk-typescript/dist/sdk/lobbyv1"`
+  } else if ($resource <: `LobbyV2Api`) {
+    $result = `"@hathora/cloud-sdk-typescript/dist/sdk/lobbyv2"`
```

Comment:
I doubt we want to import directly from dist / let's just file that as a TODO to confirm. Idiomatically most libraries prefer you to import from the root.

Reply:
This is definitely me copying the examples, but I'll double check. [random example]https://github.com/hathora/cloud-sdk-typescript/blob/main/docs/sdks/authv1/README.md#example-usage):

```js
import { HathoraCloud } from "@hathora/cloud-sdk-typescript";
import { LoginAnonymousRequest } from "@hathora/cloud-sdk-typescript/dist/sdk/models/operations";
```

 It's possible they just kept bad auto-imports in their code.

Reply:
Just confirmed they want to switch to root imports, so let's just use those.